### PR TITLE
fix: Fix 'Book a new time' link in request-reschedule for Team Event

### LINKED ISSUE #12147: [CAL-2661] "Book a New time" link in a requested reschedule email has wrong link for a team event
It is applicable to both teams inside and outside org.
https://github.com/calcom/cal.com/blob/d7c92d4e3aa50dca05718590619af6886c7b78d2/packages/core/builders/CalendarEvent/builder.ts#L247C1-L247C1

This line doesn't seem to be using the event type slug.

<sub>[CAL-2661](https://linear.app/calcom/issue/CAL-2661/book-a-new-time-link-in-a-requested-reschedule-email-has-wrong-link)</sub>

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

Fixes #12147 #12141

Also, adds tests for `requestReschedule` handler. Restructures bookingScenario test dependencies as per the requirements

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?
- Create a team event
- Book that
- Login as organizer and do a request-reschedule
- See the email that you received with 'Book a new time' link\
- Clicking on it should take you to the correct reschedule page. Earlier it was taking the user to team profile page and he was unable to reschedule.

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist
- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by hariombalhara** [comment]
Current dependencies on/for this PR:
* main
  * **PR #12261** <a href="https://app.graphite.dev/github/pr/calcom/cal.com/12261" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/calcom/cal.com/12261?utm_source=stack-comment).

File: apps/web/test/utils/bookingScenario/test.ts [hunk 1]
```diff
@@ -67,10 +67,10 @@ export const testWithAndWithoutOrg = (
   _testWithAndWithoutOrg(description, fn, timeout, "run");
 };
 
-testWithAndWithoutOrg.only = ((description, fn) => {
-  _testWithAndWithoutOrg(description, fn, "only");
+testWithAndWithoutOrg.only = ((description, fn, timeout) => {
+  _testWithAndWithoutOrg(description, fn, timeout, "only");
```
Comment (hariombalhara) [inline apps/web/test/utils/bookingScenario/test.ts:71]: It was a bug that got caught due to TS checks

File: packages/features/bookings/lib/handleNewBooking/test/fresh-booking.test.ts [hunk 2]
```diff
@@ -940,7 +939,7 @@ describe("handleNewBooking", () => {
           });
           await handleNewBooking(req);
 
-          expectBrokenIntegrationEmails({ booker, organizer, emails });
+          expectBrokenIntegrationEmails({ organizer, emails });
```
Comment (hariombalhara) [inline packages/features/bookings/lib/handleNewBooking/test/fresh-booking.test.ts:942]: Unnecessary param

File: packages/core/builders/CalendarEvent/builder.ts [hunk 3]
```diff
@@ -244,7 +248,7 @@ export class CalendarEventBuilder implements ICalendarEventBuilder {
 
       let slug = "";
       if (isTeam && eventType?.team?.slug) {
-        slug = `/team/${eventType.team?.slug}`;
+        slug = `team/${eventType.team?.slug}/${eventType.slug}`;
```
Comment (hariombalhara) [inline packages/core/builders/CalendarEvent/builder.ts:251]: / is already there in `bookerUrl` and `WEBAPP_URL`

File: apps/web/test/handlers/requestReschedule.test.ts [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+
+      expectBookingRequestRescheduledEmails({
+        booking: {
+          uid: bookingUid,
+        },
+        booker,
+        organizer: organizer,
+        loggedInUser,
+        emails,
+        bookNewTimePath: "/team/team-1/event-type-1",
```
Comment (hariombalhara) [inline apps/web/test/handlers/requestReschedule.test.ts:245]: This is the regression test.

**Review by hariombalhara — COMMENTED** [review]
Self reviewed

**Review by CarinaWolli — DISMISSED** [review]
Fix looks good, but Unit tests are failing: 

Screenshot 2023-11-10 at 13 47 12

**Comment by hariombalhara** [comment]
Thanks @CarinaWolli. Fixed the import paths that got broken due to recent merge of main

**Review by zomars — APPROVED** [review]
Nice work @hariombalhara 🙏🏽

**Pull request merged**